src/Utilities/MakeWithValue.hpp

```diff
@@ -41,17 +43,59 @@ namespace MakeWithValueImpls {
 template <typename T>
 struct MakeWithValueImpl<double, T> {
   static SPECTRE_ALWAYS_INLINE double apply(const T& /* input */,
-                                            const double value) {
+                                            const double value) noexcept {
+    return value;
+  }
+};
+
+// these fundamental type MakeWithValueImpls could be made more generic, but not
+// without causing trouble with the Tensor version of MakeWithValues. So,
+// further simplification of MakeWithValues will require SFINAE and more
+// restructuring of MakeWithValues in many places.
+/// \brief Returns a float initialized to `value` (`input` is ignored)
+template <typename T>
+struct MakeWithValueImpl<float, T> {
+  static SPECTRE_ALWAYS_INLINE float apply(const T& /* input */,
+                                           const float value) noexcept {
+    return value;
+  }
+};
+
+/// \brief Returns an int initialized to `value` (`input` is ignored)
+template <typename T>
+struct MakeWithValueImpl<int, T> {
+  static SPECTRE_ALWAYS_INLINE int apply(const T& /* input */,
+                                         const int value) noexcept {
+    return value;
+  }
+};
+
+/// \brief Returns a long initialized to `value` (`input` is ignored)
+template <typename T>
+struct MakeWithValueImpl<long, T> {
+  static SPECTRE_ALWAYS_INLINE long apply(const T& /* input */,
+                                          const long value) noexcept {
     return value;
   }
 };
 
+/// \brief Returns an arbitrary complex type initialized to `value` (`input` is
+/// ignored)
+template <typename R, typename T>
+struct MakeWithValueImpl<std::complex<R>, T> {
+  static SPECTRE_ALWAYS_INLINE std::complex<R> apply(const T& /* input */,
+                                                     const R value) noexcept {
+    return std::complex<R>(value, value);
```

Comment:
I'm happy to add tests. Is the trouble here that it would be expected to assign just the real part of the complex? The motivation for this is to have a utility that can fill all entries of a multi-field type like complex with singaling_NaN during initialization.
If this isn't fitting for `make_with_value`, I can make a separate utility to `make_signaling_NaN` to  fit my use case.

Reply:
There are places in the code (`determinant_and_inverse`, at least) that assume `make_with_value<T>(..., 1.0)` is the multiplicative identity (for `T` where that makes sense).  A separate utility seems better.

Reply:
Yea... If I created it with `1` I would expect `1 + 0 i` not `1 + i`.